Intial draft for maintainer doc (#608)
nsidc/earthaccess: docs/maintainers-guide.md

# Maintainers Guide

This page offers guidance to project maintainers regarding our setup procedures, release processes, package creation, and other related tasks.

> # Maintainer Onboarding and Best Practices

Reviewer: Let's reserve header level 1 for the title of the document, and add one extra header level to all the other sub-headers.

## Becoming a Maintainer or Triager

> If you are interested in becoming a maintainer, you can join our community. We expect you to be actively involved.

Reviewer: ```suggestion
If you are interested in becoming a maintainer, you can join our community. Maintainers have several important responsibilities, so please read on to understand the role.
```

I feel the "actively involved" part is more clearly explained in the next section :)

> Also, if you're interested in helping managing issues with labels and interacting with incoming requests, you can have a "triager" role!

Reviewer: ```suggestion
Also, if you're interested in helping managing issues with labels and interacting with incoming requests, you can have a "triager" role!

To get permissions, please start by participating on GitHub by answering questions, reviewing PRs, or contributing code or documentation. Once you're feeling comfortable, you can ask any of our maintainers for permissions by `@`ing them on GitHub.
```

2. As a maintainer, you are expected to uphold a positive and inclusive team culture. This includes following the guidelines outlined in the [Openscapes team culture page](https://openscapes.github.io/series/core-lessons/team-culture.html) and the [recorded psychological safety talk](https://www.youtube.com/watch?v=rzi-qkl8u5M) . By doing so, you can help ensure that all team members and contributors feel safe, respected, and valued.

## Maintainer Processes Beyond Regular Contributing

1. As a maintainer, label issues clearly and consistently to help contributors identify issue types and priority. Use 'good first issue' for contributor-friendly issues.

2. As a maintainer, create welcoming environment when communicating with contributors (issue / PR / discussion posters).

3. As a maintainer reviewing and merging contributions is critical. Here are some best practices:

    3a. Review contributions thoroughly.

    3b. Provide constructive feedback.

    3c. Communicate clearly and respectfully.

    3d. Merge contributions promptly.

> 4. As a maintainer, you will be releasing different versions. More on this in [here](#making-a-release).

Reviewer: The above two sections are exactly what we need! Thank you :star_struck:

Author: Could you please be more specific about this? I'm kind of confused here.

Reviewer: Sorry for being unclear. I was saying that I think the "Maintainer Responsibilities and Expectations" & " Maintainer Processes Beyond Regular Contributing" sections look great :)

# Branches

main: This is the main branch, which is consistently tested and prepared for release as a new version. Avoid pushing changes directly to this branch. Instead, create a new branch and submit a pull request for any modifications.

> gh-pages: This branch hosts HTML docs, updated by GitHub Actions. Don't commit directly. Make changes in a separate branch, then submit a pull request. GitHub Actions will automatically update docs on gh-pages when merged.

Reviewer: ```suggestion
```

Our `gh-pages` branch is static and only redirects to ReadTheDocs, so we can remove this section.

> # Continuous Integration

Reviewer: We could call this "Continuous Integration & Delivery". We have quite a few workflows going on, and I don't think we need to go in depth on all of them right now, we can always build that out more later. But I do think it's important to call out that delivery / releasing is part of our workflows :)

> The GitHub Actions CI services handle the project's building, testing, and management across Linux, macOS, and Windows platforms. The CI file manages the continuous integration processes, and list out their respective dependencies. Here we briefly summarize the functions of the files in CI.

Reviewer: ```suggestion
The GitHub Actions CI services handle the project's building, testing, and management across Linux, macOS, and Windows platforms. The CI workflow manages the continuous integration processes, and list out their respective dependencies. Here we briefly summarize the functions of the files in CI.
```

TODO: Link to the CI workflow YAML file being referenced here

> ```environment-dev.yml ```  - The environment configuration file specifies dependencies for Python development.

Reviewer: ```suggestion
`environment-dev.yml`  - The environment configuration file specifies dependencies for Python development.
```

Please change any inline code to single backticks throughout :bow:

> ```environment-mindeps.yaml``` - The test-environment file lists required and test dependencies for a testing environment.

Reviewer: ```suggestion
```environment-mindeps.yaml``` - Specifies an environment with the minimum supported dependency versions to detect backwards-incompatible code.
```

# Continuous Documentation

> The Read the Docs service is used to preview documentation changes made in pull requests. This service uses a configuration file, ```.readthedocs.yaml```.

Reviewer: ```suggestion
[ReadTheDocs](https://readthedocs.org/projects/earthaccess/) is used to generate and host [our documentation website](https://earthaccess.readthedocs.io/) as well as preview documentation changes made in pull requests. This service uses a configuration file, ```.readthedocs.yaml```.
```

# Making a Release

> To automate releases, we employ a script called ```publish.sh``` in the scripts folder. This script utilizes Poetry for managing and publishing the package to PyPI.

Reviewer: Let's re-use our existing releasing page `contributing/releasing.md` instead :)
